### ISSUE TITLE
People Directory Index doesn't use last selection

### ISSUE THREAD

**Issue by jakobnotseth** [SharePoint/sp-starter-kit#219]
#### Category
- [ ] Question
- [x] Bug
- [ ] Enhancement

#### Expected or Desired Behavior

When selecting a letter in the index, I expect the selection to filter current people in directory.

#### Observed Behavior

Selection lags one selection behind, giving the results as the previous selection.

#### Steps to Reproduce

Select letter 'B' and observe letter 'A' results (default on load), select letter 'A' and observe letter 'B' results.

#### Solution

In PeopleDirectory.tsx use index from method parameter and not from state. I.e. const selectedIndex = index and not const selectedIndex = this.state.selectedIndex in method _loadPeopleInfo

**Comment by eoverfield** [comment]
@jakobnotseth great catch! That is quite correct, because of the async nature of React's setState method, we cannot be sure that state will be settled by the time we use this.state... to retrieve a state again.

Rather than update _loadPeopleInfo, I suggest instead that we change the way we re-call _loadPeopleInfo after an event. I will look to make a PR that changes our call strategy to something similar to:

```
this.setState({
  selectedIndex: index,
  searchQuery: ''
},
function() {
    // load information about people matching the selected tab
    this._loadPeopleInfo(index, null);
}
);
```

setState allows for a second optional parameter, a function that is called after state is actually set. At that point we are confident we can get the state that we set.

There is an argument that the setState callback is not preferred, rather we should use the React lifecycle and shouldComponentUpdate. I find the callback works well in these cases, and would be easier than refactoring this code at this point.

**Comment by eoverfield** [comment]
@jakobnotseth addressed in PR https://github.com/SharePoint/sp-starter-kit/pull/220. If you have a moment, you might want to pull this update and see if it fixes the issue for you. In my testing, worked as expected.

**Comment by jakobnotseth** [comment]
Works as expected in my tests as well :)

**Comment by bcameron1231** [comment]
Closing - Fixed via #220. Confirmed by Author.
